src/rust/engine/process_execution/src/lib.rs

```diff
@@ -91,6 +112,38 @@ pub struct ExecuteProcessRequest {
   pub jdk_home: Option<PathBuf>,
 }
 
+impl TryFrom<MultiPlatformExecuteProcessRequest> for ExecuteProcessRequest {
+  type Error = String;
+
+  fn try_from(req: MultiPlatformExecuteProcessRequest) -> Result<Self, Self::Error> {
+    match req.0.get(&(Platform::None, Platform::None)) {
+      Some(crossplatform_req) => Ok(crossplatform_req.clone()),
+      None => Err(String::from(
+        "Cannot coerce to a simple ExecuteProcessRequest, no cross platform request exists.",
+      )),
+    }
+  }
+}
+
+///
+/// A container of platform constrained processes.
+///
+#[derive(Derivative, Clone, Debug, Eq)]
+#[derivative(PartialEq, Hash)]
```

Comment:
Why not just `derive` these? It looks like they're not being customised?

Reply:
I think I just copied this pattern from another struct. I'll change them to regular derive's